pkg/arvo/sys/vane/ames.hoon

```diff
@@ -76,5405 +76,8485 @@
-      ++  trace-ames     (cury trace %ames)
-      ++  ev-trace
-        |=  [verb=? =ship print=(trap tape)]
-        ^+  same
-        (trace-ames verb ship ships.bug.ames-state print)
-      ::  +get-peer-state: lookup .her state or ~
-      ::
-      ++  get-peer-state
-        |=  her=ship
-        ^-  (unit peer-state)
-        ::
-        =-  ?.(?=([~ %known *] -) ~ `+.u)
-        (~(get by peers.ames-state) her)
-      ::  +got-peer-state: lookup .her state or crash
-      ::
-      ++  got-peer-state
-        |=  her=ship
-        ^-  peer-state
-        ::
-        ~|  %freaky-alien^her
-        =-  ?>(?=(%known -<) ->)
-        (~(got by peers.ames-state) her)
-      ::  +gut-peer-state: lookup .her state or default
-      ::
-      ++  gut-peer-state
-        |=  her=ship
-        ^-  peer-state
-        =/  ship-state  (~(get by peers.ames-state) her)
-        ?.  ?=([~ %known *] ship-state)
-          *peer-state
-        +.u.ship-state
-      ::
-      ++  get-sponsors
-        ;;  (list ship)
-        =<  q.q  %-  need  %-  need
-        (rof [~ ~] /ames %j `beam`[[our %saxo %da now] /(scot %p our)])
-      ::
-      +|  %tasks
-      ::  +on-take-flub: vane not ready to process message, pretend it
-      ::                 was never delivered
-      ::
-      ++  on-take-flub
-        |=  =wire
-        ^+  event-core
-        ?~  parsed=(parse-bone-wire wire)
-          ::  no-op
-          ::
-          ~>  %slog.0^leaf/"ames: dropping malformed wire: {(spud wire)}"
-          event-core
-        ?>  ?=([@ her=ship *] u.parsed)
-        =*  her  her.u.parsed
-        =/  peer-core  (abed-got:pe her)
-        ?:  ?&  ?=([%new *] u.parsed)
-                (lth rift.u.parsed rift.peer-state.peer-core)
-            ==
-          ::  ignore events from an old rift
-          ::
-          %-  %^  ev-trace  odd.veb  her
-              |.("dropping old rift wire: {(spud wire)}")
-          event-core
-        =/  =bone
-          ?-(u.parsed [%new *] bone.u.parsed, [%old *] bone.u.parsed)
-        abet:(on-flub:peer-core bone)
-      ::  +on-take-done: handle notice from vane that it processed a message
-      ::
-      ++  on-take-done
-        |=  [=wire error=(unit error)]
-        ^+  event-core
-        ?~  parsed=(parse-bone-wire wire)
-          ::  no-op
-          ::
-          ~>  %slog.0^leaf/"ames: dropping malformed wire: {(spud wire)}"
-          event-core
-        ?>  ?=([@ her=ship *] u.parsed)
-        =*  her        her.u.parsed
-        =/  peer-core  (abed-got:pe her)
-        |^
-        ?:  ?&  ?=([%new *] u.parsed)
-                (lth rift.u.parsed rift.peer-state.peer-core)
-            ==
-          ::  ignore events from an old rift
-          ::
-          %-  %^  ev-trace  odd.veb  her
-              |.("dropping old rift wire: {(spud wire)}")
-          event-core
-        =/  =bone
-          ?-(u.parsed [%new *] bone.u.parsed, [%old *] bone.u.parsed)
-        =?  peer-core
```

Comment:
@yosoyubik @lukechampine I think the lss problem is here -- this should be `(snag ?((gth wid 4) fag (dec fag)) pairs.lss-proof)`.